### PR TITLE
[SPARKTA-219] Cube functions for updates only the dimensions changed

### PR DESCRIPTION
**Pull request by compae**
#### Description

New functions in Cube in order to mark the dimensionsValues that are changed on the actual streaming window, if are been changed this values must be upserted in the Output. 
SPARKTA-219
#### Reviewer

@anistal   
#### Test

All passed.
#### Coverage

No changes
#### Scalastyle

No info

### CODE REVIEW

**Comment by anistal** [comment]
+1

**Comment by sgomezg** [comment]
+1

**Comment by gserranojc** [comment]
Added refactor in Cube for better comprehension. Added some case clases.

**Comment by sgomezg** [comment]
+1.
Please squash the commits

**Comment by gserranojc** [comment]
Amend maked!

Added the changes that @danielcsant said.

**Comment by danielcsant** [comment]
test-at fail

**Comment by gserranojc** [comment]
Removed Redis test, we have conflicts with others test in test-at. This is less important than this Pull Request.
I comment this test, we open one issue for this proposal.

**Comment by danielcsant** [comment]
test-at module does not compile

File: sdk/src/main/scala/com/stratio/sparkta/sdk/Output.scala [hunk 1]
```diff
@@ -58,19 +58,19 @@ abstract class Output(keyName: String,
     case Some(fixDimensions) => fixDimensions.split(FieldsSeparator)
   }
 
-  val fixedAgg = properties.getString("fixedAggregation", None)
+  val fixedMeasure = properties.getString("fixedMeasures", None)
```
Comment (danielcsant) [inline sdk/src/main/scala/com/stratio/sparkta/sdk/Output.scala]: fixedAggregation appears several times in the module. If you do this refactor, rename every reference to this property and update documentation https://stratio.atlassian.net/wiki/display/SPARKTA0x8/

**Pull request closed without merging**